src/main/kotlin/io/libp2p/etc/types/OtherExt.kt

```diff
@@ -36,3 +38,6 @@ fun <T> defer(f: (Deferrable) -> T): T {
         deferrable.execute()
     }
 }
+
+fun <T : Throwable> Throwable.hasCauseOfType(clazz: KClass<T>): Boolean =
+    clazz.isInstance(this) || (cause != null && cause!!.hasCauseOfType(clazz))
```

Comment:
Isn't it possible to get stuck in an infinite loop here?  In [teku we're using a util](https://github.com/ConsenSys/teku/blob/b7fae2614f5a4e9e0e0c50706e9ad6790e680357/util/src/main/java/tech/pegasys/teku/util/exceptions/ExceptionUtil.java#L22-L28) that handles recursive causes.

Reply:
Yeah, good point. 
I heard something of such weird cases with looped `Throwable` causalities and both utils are handling those cases.